app/models/user.rb

```diff
@@ -1,4 +1,6 @@
 class User < ActiveRecord::Base
+  attr_accessor :asserted_attributes
```

Comment:
Or can you just pass asserted_attributes to SamlIdp rather than the User itself, and make sure phone and email are part of that?